phc/easy/query.py

```diff
@@ -49,43 +79,31 @@ def execute_dsl(
           "type": "select",
           "columns": "*",
           "from": [
-              {
-                  "table": "observation"
-              }
+              {"table": "patient"}
           ],
-          "limit": [
-              {
-                  "type": "number",
-                  "value": 0
-              },
-              {
-                  "type": "number",
-                  "value": 1000
-              }
-          ]
-        }, scroll=True)
+        }, all_results=True)
         """
-        auth = Auth(auth_args)
-        fhir = Fhir(auth.session())
-
-        response = fhir.execute_es(
-            auth.project_id,
-            query,
-            _scroll_id if query_allows_scrolling(query) and scroll else "",
-        )
-
-        current_results = response.data.get("hits").get("hits")
-        results = [*_prev_hits, *current_results]
-        _scroll_id = response.data.get("_scroll_id", "")
-
-        if len(current_results) == 0 or scroll is False:
-            return results
-
-        # TODO: Revisit private parameters being exposed in public method
-        return Query.execute_dsl(
-            query,
-            scroll=True,
-            auth_args=auth,
-            _scroll_id=_scroll_id,
-            _prev_hits=results,
-        )
+        if all_results:
+            return recursive_execute_dsl(
+                {
+                    **query,
+                    "limit": [
+                        {"type": "number", "value": 0},
+                        # Make window size smaller than maximum to reduce
+                        # pressure on API
+                        {"type": "number", "value": int(MAX_RESULT_SIZE / 5)},
+                    ],
+                },
+                all_results,
+                auth_args,
+            )
+
+        # Scroll if limit is above MAX_RESULT_SIZE
+        limit = iter(query.get("limit", []))
+
+        lower = next(limit, {}).get("value", 0)
+        upper = next(limit, {}).get("value", 0)
+
+        scroll = True if upper - lower > MAX_RESULT_SIZE else all_results
```

Comment:
Minor: can directly assign the comparison result, i.e. `scroll = upper -  lower > MAX_RESULT_SIZE`